src/harness/sas_interface.py

```diff
@@ -519,5 +519,58 @@ def TriggerDailyActivitiesImmediatelyAndWaitUntilComplete(self):
     If the status is not changed within 2 hours it will throw an exception.
     """
     pass
+
+  @abc.abstractmethod
+  def TriggerCreateActivityDump(self):
+    """
+    Trigger SAS to create a full activity dump
+    """
+    pass
+
+  def InjectPeerSas(self, request):
+    """
+        Inject peer SAS
+    """
+    pass
+
+
+class HttpServerInterface(object):
+  @abc.abstractmethod
+  def setupServer(self, parameters=None):
```

Comment:
Why not take each parameter as a named var rather than as a dictionary? Since the server will be a local Python object, I think the latter is the standard approach.

Reply:
Parameter is used to store the response body of SAS test harness server. To make the parameter as the dict will better to do the response. The schema of parameter will be: 

{ "full_activity" : dict to store full activity dump objects,
   "cbsd_activity": dict to store cbsd activity dump objects,
   "PPA_activity" : dict to store ppa activity dump objects,
   "esc_activity" : dict to store esc activity dump objects
   }
   schema of full_activity {full activity dump object}
   schema of cbsd_activity {"id": {cbsd activity dump object}}
   schema of PPA_activity {"id": {ppa activity dump object}}
   schema of esc_activity {"id": {esc activity dump object}}

Reply:
I'm quoting your response and replying here so that we can keep a single thread going.

> Already changed the name from server_details to baseUrl_portNumber. Because the server_details is {"baseurl": url, "port": portnumber}

Please pass them as individual arguments. The convention you're proposing is very non-standard and error-prone.